client/src/util/centroid.js

```diff
@@ -6,100 +6,131 @@ import { unassignedCategoryLabel } from "../globals";
   Centroid coordinate calculation
```

Comment:
I would add a prominent comment that this calculates centroids for a truncated set of labels.  Ie, it follows the display logic of truncating categories that have excessive labels.

it is not a general purpose centroid calculator, as it will ignore labels in those categorical annotations that have more than the current system truncation setting.

(this is a feature, but also means that this code is only useful for display purposes that want the centroids for the list of labels we will show the user in the LSB)